### PR TITLE
[SPARK-50358][SQL][TESTS] Update postgres docker image to 17.1

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
This PR aims to update the `postgres` docker image from `17.0` to `17.1`.

### Why are the changes needed?
This will help Apache Spark test the latest postgres.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Review by yaooqinn — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

File: connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/PostgresNamespaceSuite.scala [hunk 1]
```diff
@@ -26,16 +26,16 @@ import org.apache.spark.sql.util.CaseInsensitiveStringMap
...
- *   ENABLE_DOCKER_INTEGRATION_TESTS=1 POSTGRES_DOCKER_IMAGE_NAME=postgres:17.0-alpine
+ *   ENABLE_DOCKER_INTEGRATION_TESTS=1 POSTGRES_DOCKER_IMAGE_NAME=postgres:17.1-alpine
  *     ./build/sbt -Pdocker-integration-tests "testOnly *v2.PostgresNamespaceSuite"
  * }}}
  */
 @DockerTest
 class PostgresNamespaceSuite extends DockerJDBCIntegrationSuite with V2JDBCNamespaceTest {
   override val db = new DatabaseOnDocker {
-    override val imageName = sys.env.getOrElse("POSTGRES_DOCKER_IMAGE_NAME", "postgres:17.0-alpine")
+    override val imageName = sys.env.getOrElse("POSTGRES_DOCKER_IMAGE_NAME", "postgres:17.1-alpine")
```
[the remark this answers is not in the document]
Comment (yaooqinn) [reply connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/PostgresNamespaceSuite.scala:38]: We can have a PostgresDatabaseOnDocker class just like OracleDatabaseOnDocker, DB2DatabaseOnDocker, etc.

[the remark this answers is not in the document]
Comment (cloud-fan) [reply connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/PostgresNamespaceSuite.scala:38]: +1

[the remark this answers is not in the document]
Comment (panbingkun) [reply connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/PostgresNamespaceSuite.scala:38]: If no one has started yet, I'll do this.

[the remark this answers is not in the document]
Comment (panbingkun) [reply connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/PostgresNamespaceSuite.scala:38]: https://github.com/apache/spark/pull/49004

**Pull request closed without merging**